lib/services/elasticsearch.js

```diff
@@ -919,162 +949,123 @@ class ElasticSearch extends Service {
+        const
+          esIndexes = body.map(({ index }) => index),
+          indexes = this._extractIndexes(esIndexes),
+          apiResponse = {
+            indexes
+          };
 
-        return {collections: {stored: collections}};
+        return apiResponse;
       })
-      .catch(error => Bluebird.reject(
-        this.esWrapper.formatESError(error))
-      );
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
    * Reset all indexes that the users is allowed to delete
    *
-   * @param {Request} request
-   * @return {Promise}
+   * @param {string} indexes - Index names
+   *
+   * @return {Promise} { deleted }
    */
-  deleteIndexes(request) {
-    const deletedIndexes = request.input.body.indexes;
+  deleteIndexes (indexes) {
+    let deleted = [];
 
-    if (deletedIndexes === undefined || deletedIndexes.length === 0) {
-      return Bluebird.resolve({deleted: []});
-    }
+    return this.client.cat.indices({ format: 'json' })
+      .then(({ body }) => {
+        const
+          esIndexes = body
+            .map(({ index }) => index)
+            .filter(esIndex => {
+              const index = this._extractIndex(esIndex);
 
-    return this.client.indices.delete({index: deletedIndexes})
-      .then(() => ({deleted: deletedIndexes}))
-      .catch(error => Bluebird.reject(
-        this.esWrapper.formatESError(error))
-      );
-  }
+              if (esIndex[0] !== this.indexPrefix || ! indexes.includes(index)) {
+                return false;
+              }
 
-  /**
-   * List all known indexes
-   *
-   * @returns {Promise}
-   */
-  listIndexes() {
-    return this.client.indices.getMapping()
-      .then(result => ({
-        indexes: Object.keys(result).filter(
-          indexName => indexName !== '' && indexName[0] !== '%')
-      }))
-      .catch(error => Bluebird.reject(
-        this.esWrapper.formatESError(error))
-      );
-  }
+              deleted.push(index);
 
-  /**
-   * Create a new index
-   *
-   * @param {Request} request
-   * @returns {Promise}
-   */
-  createIndex(request) {
-    const index = getIndex(request);
+              return true;
+            }),
+          esRequest = {
+            index: esIndexes
+          };
 
-    return this.client.indices.create({index})
-      .catch(error => Bluebird.reject(
-        this.esWrapper.formatESError(error))
-      );
+        return this.client.indices.delete(esRequest);
+      })
+      .then(() => {
+        const apiResponse = {
+          deleted
+        };
+
+        return apiResponse;
+      })
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
    * Delete an index
    *
-   * @param {Request} request
-   * @returns {Promise}
+   * @param {string} index - Index name
+   *
+   * @returns {Promise} { }
    */
-  deleteIndex(request) {
-    const index = getIndex(request);
+  deleteIndex (index) {
+    return this.deleteIndexes([index])
+      .then(() => {
+        const apiResponse = {};
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "apiResponse". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1488)